docs/develop/validation.md

```diff
@@ -0,0 +1,86 @@
+# Running validation
+
+
+!!! note "Currenlty, there is validation only for retrieval, i.e. `POST /embeddings-search` endpoint."
+
+To evaluate the performance your model (along with your own configurations and
+guardrails), run the validation test(s) in `core_backend/validation`.
+
+The validation is written in pytest so that we can call the endpoint.
+
+## Retrieval (`/embeddings-search`) validation
+
+We evaluate the "performance" of retrieval by computing "Top K Accuracy", which we
+define as proportion of times the best matching answer was present in top K retrieved content.
+
+### Preparing the data
+The test assumes the validation data contains a single label representing the best
+matching content, rather than a ranked list of all relevant content.
+
+An example validation data will look like
+|query|label|
+|--|--|
+|"How?"|0|
+|"When?"|1|
+|"What year was it?"|1|
+|"May I?"|2|
+
+An example content data will look like
+|content_text|label|
+|--|--|
+|"Here's how."|0|
+|"It was 2024."|1|
+|"Yes"|2|
+
+
+### How to run
+
+1. Create a new python environment:
+    ```shell
+    conda create -n "aaq-validate" python=3.10
+    ```
+    You van also copy the existing `aaq-core` environment.
+2. Install requirements. This assumes you are in project root `aaq-core`.
+    ```shell
+    conda activate aaq-validate
+    pip install -r core_backend/requirements.txt
+    pip install -r core_backend/validation/requirements.txt
+    ```
+3. Set environment variables.
+   The following is required:
+    ```
+    conda env config vars set OPENAI_API_KEY=<Your OPENAI API key>
+    ```
+    For optional ones, check out the defaults in `core_backend/app/configs/app_config.py`
+    and modify as per your own requirements:
+    ```
+    conda env config vars set EMBEDDING_MODEL=<...>
+    ```
+5. Check the environment variables defined in `core_backend/validation/validation.env`.
```

Comment:
Do we really need to check this or defaults just work? Also, what sets up the qdrant container? Is it assumed that there is one already running?

Can I suggest that we just put all of these in a Makefile target like we do for `make tests`. Though i know passing argument is a real pain.

Reply:
Yeah parsing command line arguments for make target doesn't seem very straightforward.. 

I will not make this because I don't see too much value in making the make target. It's a one-time job to export the required variables to the conda env, and if users are using external API they need to remember to export the token, too (I have clarified this in the docs now!).